### DISCUSSION TITLE
Tkinter-利用Python建立GUI

### DISCUSSION THREAD

**Discussion by s96015welson** [1101-GS4719-C/Python-Community-GS4719-C#1093] [優秀Python套件]
tkinter是python中一個幫忙製作GUI(圖形使用者介面)的好幫手，裡面提供了許多物件像是按鈕、視窗、文字標籤、捲軸等等，儘管tkinter較樸素、變化性較少，卻很好上手適合入門，有興趣的同學可以嘗試做做看。

另外有幾點需要注意的地方:
1.盡量使用gird或full，避免使用place，因為前兩者會隨著視窗縮放而自動改變配置，而後者在放大縮小視窗時可能會出現顯示不完全的情形。
2.可以善用config()來更改已經建立好的元件的參數。
3.要注意結束視窗的撰寫，有時候直接關閉視窗會造成程式其他部分執行錯誤，使得程式卡死，可以搭配異常處理(try...except)來避免這問題發生。

Tkinter-利用Python建立GUI(基本操作及佈局篇):https://ithelp.ithome.com.tw/articles/10278264
Tkinter-利用Python建立GUI(元件篇):https://ithelp.ithome.com.tw/articles/10278708

**Comment by RogelioKG** [comment]
❄️冷知識：
IDLE 是用 tkinter 寫的！
要建立 GUI，tkinter 算是一個比較輕量的選擇。
不過就生態完整度的話，PyQt 還是更有優勢。
即便如此，tkinter 很簡單很方便，仍然是一個值得學習的模組。
